### PR TITLE
For #476: implementation of starlight card

### LINKED ISSUE #476: Create Astromancer card Starlight, its effects and tests
**Starlight** 
Cost: 1 Mana 
Effect: Deal 5 to all enemies. 
Type: Offensive
Feed-Solar: +10% spell Power for each enemy hit by Starlight.

### PR DESCRIPTION
**Pull request by paulodamaso**
Closes #476: implementation of starlight card

### CODE REVIEW

**Comment by paulodamaso** [comment]
@SilasReinagel Sorry I misunderstood the description: I thought that each enemy hit should get +10% spellpower. Will fix it

**Comment by paulodamaso** [comment]
@SilasReinagel Finished the implementation adding the following:
- counter logic of how many targets were affected by the last card
- created a `FloatVariable` representing the multiplier to be applied, based on previously stored value

File: src/ProjectNeon/Assets/Features/Effects/Events/SpellToPerform.cs [hunk 1]
```diff
@@ -0,0 +1,11 @@
+﻿
+public sealed class SpellToPerform
```
Comment (SilasReinagel) [inline src/ProjectNeon/Assets/Features/Effects/Events/SpellToPerform.cs]: If we are going to have abstract events for Spells, they can't depend on concrete spell effects.

File: src/ProjectNeon/Assets/Data/Templates/Editor/GameDataMenu.cs [hunk 2]
```diff
@@ -18,6 +18,15 @@ static class GameDataMenu
     [MenuItem("Assets/Create/GameContent/DumbAI")]
     static void DumbAI() => Create<DumbAI>();
 
+    [MenuItem("Assets/Create/Variable/Spell Hits Last Card Multiplier")]
```
Comment (SilasReinagel) [inline src/ProjectNeon/Assets/Data/Templates/Editor/GameDataMenu.cs]: Don't leave editor menu items for things that will only be created once. 

Just create one and then kill the editor code.

File: src/ProjectNeon/Assets/Features/Battle/Cards/CardResolutionZone.cs [hunk 3]
```diff
@@ -9,7 +9,8 @@ public class CardResolutionZone : ScriptableObject
     [SerializeField] private CardPlayZone physicalZone;
     [SerializeField] private CardPlayZone playedDiscardZone;
     [SerializeField] private GameEvent onFinished;
-    
+    public int SpellHitsOnLastCard { get; set; }
```
Comment (SilasReinagel) [inline src/ProjectNeon/Assets/Features/Battle/Cards/CardResolutionZone.cs]: This does not belong here. 

Realistically, we shouldn't be changing this class, basically ever.

File: src/ProjectNeon/Assets/Features/Battle/Variables/SpellHitsLastCardMultiplier.cs [hunk 4]
```diff
@@ -0,0 +1,11 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class SpellHitsLastCardMultiplier : FloatVariable
+{
+    [SerializeField] CardResolutionZone resolutionZone;
```
Comment (SilasReinagel) [inline src/ProjectNeon/Assets/Features/Battle/Variables/SpellHitsLastCardMultiplier.cs]: Number variables should not have any external dependencies.

File: src/ProjectNeon/Assets/Features/Battle/Variables/SpellHitsLastCardMultiplier.cs [hunk 5]
```diff
@@ -0,0 +1,11 @@
...
+
+    public new float Value => resolutionZone.SpellHitsOnLastCard * 0.1f;
```
Comment (SilasReinagel) [inline src/ProjectNeon/Assets/Features/Battle/Variables/SpellHitsLastCardMultiplier.cs]: This doesn't work. No user of the FloatVariable will access this value.

File: src/ProjectNeon/Assets/Features/Effects/Events/SpellPerformed.cs [hunk 6]
```diff
@@ -0,0 +1,11 @@
+﻿
+public sealed class SpellPerformed
+{
+    public SpellFlatDamageEffect SpellFlatDamageEffect { get; }
+
+    public SpellPerformed(SpellFlatDamageEffect spell)
```
Comment (SilasReinagel) [inline src/ProjectNeon/Assets/Features/Effects/Events/SpellPerformed.cs]: If we are going to have abstract events for Spells, they can't depend on concrete spell effects.

File: src/ProjectNeon/Assets/Features/Battle/Variables/SpellHitsLastCardMultiplier.cs (hunk 4 above)
Comment (paulodamaso) [reply src/ProjectNeon/Assets/Features/Battle/Variables/SpellHitsLastCardMultiplier.cs]: @SilasReinagel But then, how could I acces this value from the unity editor? How could I project this value onto something if I can't makeit visible from insie this variable?

File: src/ProjectNeon/Assets/Features/Battle/Variables/SpellHitsLastCardMultiplier.cs (hunk 5 above)
Comment (paulodamaso) [reply src/ProjectNeon/Assets/Features/Battle/Variables/SpellHitsLastCardMultiplier.cs]: @SilasReinagel Why not? It is accesed the same way `FloatVariable`

File: src/ProjectNeon/Assets/Features/Battle/Cards/CardResolutionZone.cs (hunk 3 above)
Comment (paulodamaso) [reply src/ProjectNeon/Assets/Features/Battle/Cards/CardResolutionZone.cs]: @SilasReinagel Okay. How about putting it onto `BattleState` then?

File: src/ProjectNeon/Assets/Features/Effects/Events/SpellToPerform.cs (hunk 1 above)
Comment (paulodamaso) [reply src/ProjectNeon/Assets/Features/Effects/Events/SpellToPerform.cs]: @SilasReinagel Fixed it, made it to depend on generic Effect

File: src/ProjectNeon/Assets/Features/Effects/Events/SpellPerformed.cs (hunk 6 above)
Comment (paulodamaso) [reply src/ProjectNeon/Assets/Features/Effects/Events/SpellPerformed.cs]: @SilasReinagel Fixed it, made it to depend on generic Effect

**Comment by paulodamaso** [comment]
@SilasReinagel I've fixed some of the problems, but I have some questions about the other comments, please take a look

**Pull request closed without merging**